libraries/Matter/src/devices/DevicePowerSource.h

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "MatterDevice.h"
+
+class DevicePowerSource : public Device
+{
+public:
+  enum Changed_t{
+    kChanged_BatPercentRemaining = kChanged_Last << 1,
```

Comment:
This could be extended for other attributes, e.g. for voltage and alarm reporting.